CMakeLists.txt

```diff
@@ -6,14 +6,36 @@ if(CCACHE_PROGRAM)
   message(STATUS "ccache enabled (${CCACHE_PROGRAM})")
 endif()
 
+set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+
+##################################
+## setup Hunter and its modules ##
+##################################
+option(HUNTER_ENABLED "Enable Hunter package manager" OFF)
+
+set(HUNTER_ROOT "${CMAKE_SOURCE_DIR}/external" CACHE PATH "Hunter root directory")
```

Comment:
hunter root should be outside of the project, because other projects can potentially depend on the same hunter dependencies, and in this case, they can re-use them. 

Other projects can reuse packages from `external` only *if their hunter root is set to iroha's external folder*, which is bad, because other projects may not depend on (or know anything about) iroha.

Reply:
Iroha by itself is not a library. At least in the current product vision, so in this case, we shouldn't care about projects that potentially will use Iroha.

Reply:
@muratovv you did not get. 

Here is the example:
iroha uses gtest and iroha-ed25519.
iroha-ed25519 uses gtest.

With HUNTER_ROOT outside of the project, gtest will be downloaded once and reused by both iroha-ed25519 and iroha.

Reply:
If user wants to use HUNTER_ROOT in his system and share it with other projects, he can define it with `-DHUNTER_ROOT`. Environment variables are not as obvious as CMake parameters, so why use environment variable instead of CMake parameter?

Reply:
Regarding iroha-ed25519, we can define HUNTER_ROOT of iroha-ed25519 to point to iroha external in configuration step.

Reply:
@lebdron that is the point. If you do so, you need to define HUNTER_ROOT=`iroha's external` inside `iroha-ed25519`. With this example it is possible, because we can push to `iroha-ed25519`. What if there is another dependency, like grpc, which also uses gtest? We can't set HUNTER_ROOT for grpc.

Reply:
>HUNTER_ROOT should NOT be overrided in CMake at any cost
But it is not overridden because it is `CACHE`. User can safely override it with `-D`. Iroha dependencies will also have the value transitively.